app/controllers/lightweight_activities_controller.rb

```diff
@@ -47,7 +57,11 @@ def show # show index
     @run.increment_run_count!
 
     if @activity.layout == LightweightActivity::LAYOUT_SINGLE_PAGE
-      redirect_to activity_single_page_with_run_path(@activity, @run.key) and return
+      if @run.sequence
+        redirect_to sequence_activity_single_page_with_run_path(@run.sequence, @activity, @run.key) and return
+      else
+        redirect_to activity_single_page_with_run_path(@activity, @run.key) and return
+      end
```

Comment:
More lost `mode=teacher-edition`.  Currently teacher edition is not handled by single_page activities. But as long as this code is change it seems it should be supported.  However it might not as simple as just passing the current parameters so it is worth testing this out.

Reply:
In this case we were not passing the request.query_parameters before, but now we are.  I don't think this is is a problem though.